src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,321 @@
+package duke;
+
+import java.util.ArrayList;
+import duke.exceptions.InvalidCommandException;
+import duke.exceptions.InvalidFileException;
+import duke.exceptions.InvalidTaskIndexException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+
+class Duke {
+    private static final int LINE_CHAR_NUM = 40;
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static Scanner scan = new Scanner(System.in);
+    private static final String DIR_PATH = "data/";
+    private static final String FILE_PATH = "data/save.txt";
+
+    public static void main(String[] args) {
+        try {
+            loadFile();
+        } catch (InvalidFileException exception) {
+            System.out.println("Invalid save file detected.");
+        }
+        String text;
+
+        drawHorizontalLine(LINE_CHAR_NUM);
+        showLogo();
+        drawHorizontalLine(LINE_CHAR_NUM);
+        greet();
+        drawHorizontalLine(LINE_CHAR_NUM);
+
+        do {
+            text = readInput();
+            echo(text);
+        } while (!text.equals("bye"));
+
+    }
+
+    private static String readInput() {
+        return scan.nextLine();
+    }
+
+    private static void printSuccessfulAddTaskMessage(Task task) {
+        System.out.println(String.format("Got it. I've added this task to your list:\n%s", task));
+        countTaskNumber();
+    }
+
+    private static void addToDo(String description, boolean isPrintingMessage) throws InvalidCommandException {
+        description = description.trim();
+        if (description.equals("")) throw new InvalidCommandException(1);
```

Comment:
For this `if` block, you are missing `{` and `}`. 

It should look like this:

`if (description.equals("")) {
    throw new InvalidCommandException(1);
}`

Reply:
The same applies to all other instances in your other functions as well.